### PR TITLE
fix(storage-account)!: load location and tags from the resource group 

### PR DESCRIPTION
**Pull request by fargozhu**
## Description

This pull request goal is to bring consistent between all azure resources when setting both location and tags, by loading that data from the resource group data block instead of handling as an input parameter.

## PR Checklist

Please ensure the following before submitting this PR:

- [x] You complied with the code style of this project.
- [x] You formatted all Terraform configuration files to a canonical format (Hint: `terraform fmt`).
- [x] You validated all Terraform configuration files (Hint: `terraform validate`).
- [x] You executed a speculative plan, showing what actions Terraform would take to apply the current configuration (Hint: `terraform plan`).
- [x] The documentation was updated (`README.md`, `CHANGELOG.md`, etc.).
- [x] Whenever possible, the dependencies were updated to the latest compatible versions.

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix.
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no interface changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes.
- [ ] Other... Please describe:

### What is the new behavior?

There's no longer the need to specify the storage account, and both resource group and storage accounts are merged and duplications removed. 

### How to test it

1. clone the repo and checkout to the fix branch.
2. copy the follow code snippet to the _main.tf_
```hcl
module "rg" {
  source = "git::github.com/Nmbrs/tf-modules//azure/resource_group?ref=main"
  name        = "Heimdall"
  country     = "NL"
  environment = "dev"
  squad       = "Infra"
  product     = "Microservice Platform"
  extra_tags = {
    MonitoredOn = "Datadog"
    Version     = "1.0"
  }
}

module "storage_account" {
  source              = "~/tf-modules/azure/storage_account"
  name                = "saheimdall123"
  resource_group_name = module.rg.name
  extra_tags = {
    Version : "1.0"
    Visibility : "internal"
  }

  account_kind     = "Storage"
  account_tier     = "Standard"
  replication_type = "GRS"
}
```

3. execute the follow terraform commands:
```shell
$ terraform init -upgrade && terraform validate && terraform fmt --recursive && terraform plan
```

### Does this PR introduce a breaking change?

- [x] Yes.
- [ ] No.

### CODE REVIEW

**Review by pmmluis — APPROVED** [review]

File: azure/storage_account/variables.tf [hunk 1]
```diff
@@ -18,37 +18,19 @@ variable "name" {
...
 variable "resource_group_name" {
-  description = "The name of an existing Resource Group."
   type        = string
-
-  validation {
-    condition     = can(coalesce(var.resource_group_name))
-    error_message = "The 'resource_group_name' value is invalid. It must be a non-empty string."
-  }
+  description = "The name of an existing Resource Group."
 }
```
Comment (filipefigueredo) [inline azure/storage_account/variables.tf:24]: Please, consider keeping the `resource_group_name` validation ensuring that the variable is a non-empty string as we do in the `name` variable
```suggestion
variable "resource_group_name" {
  description = "The name of an existing Resource Group."
  type        = string

  validation {
    condition     = can(coalesce(var.resource_group_name))
    error_message = "The 'resource_group_name' value is invalid. It must be a non-empty string."
  }
}
```

**Review by filipefigueredo — CHANGES_REQUESTED** [review]

File: azure/storage_account/variables.tf (hunk 1 above)
Comment (pmmluis) [reply azure/storage_account/variables.tf:24]: But this is not necessary since it will always check the name given from the RG. From the data.

Comment (fargozhu) [reply azure/storage_account/variables.tf]: agree...

File: azure/storage_account/README.md [hunk 2]
```diff
@@ -15,11 +15,10 @@ Fundamentally, you need to declare the module and pass the following variables i
...
 ```hcl
 module "storage_account" {
-  source              = "../tf-modules/azure/storage-account"
+  source              = "git::github.com/Nmbrs/tf-modules/azure/storage_account?ref=v5.1.0"
   name                = "sauniquename123"
-  location            = "westeurope"
   resource_group_name = "rg-my-resource-group"
-  tags = {
+  extra_tags = {
     my_tag : "my_tag_value"
```
Comment (fargozhu) [inline azure/storage_account/README.md:22]:
```suggestion
    version: "1.0"
```

**Review by filipefigueredo — APPROVED** [review]

File: azure/storage_account/README.md (hunk 2 above)
Comment (fargozhu) [reply azure/storage_account/README.md:22]: this was suggested to show the github feature during the GitHub Top 10 session.

**Pull request merged**